### PR TITLE
dashboards daclaration

### PR DESCRIPTION
**Pull request by SergeyPl4ik**

### CODE REVIEW

File: api/controller/dashboard.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            data.circulatingSupply = await this.rxBus.get('ACCOUNT_GET_CIRCULATING_SUPPLY', req);
+            data.account = await this.rxBus.get('ACCOUNT_GET', req);
+            data.transactions = await this.rxBus.get('TRANSACTION_GET', req);
+            data.unconfirmedTransaction = await this.rxBus.get('TRANSACTION_UNCONFIRMED', req);
+            data.countStakeholders = await this.rxBus.get('STAKE_ORDERS_COUNT_STAKEHOLDERS', req);
+            data.accountsCount = await this.rxBus.get('ACCOUNT_COUNT', req);
+            data.forgingStatus = await this.rxBus.get('DELEGATE_FORGING_STATUS', req);
+            data.totalDDKStaked = await this.rxBus.get('STAKE_ORDERS_GET', req);
+
+            return new ResponseEntity({
```
Comment (alex-arugon) [inline api/controller/dashboard.ts]: Should be called after `try - catch` block

File: api/controller/dashboard.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                data: new Dashboard(data)
+            });
+        } catch (error) {
+            return new ResponseEntity({
+                data: new Dashboard(data),
```
Comment (alex-arugon) [inline api/controller/dashboard.ts]: There is no necessity to create new entity, just return `data`

File: api/controller/dashboard.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    constructor () {
+    }
+
+    @GET('/get')
+    public async getDashBoard (req: IRequest): Promise<ResponseEntity<Dashboard>> {
+        let data = new Dashboard();
+
+        try {
+            // todo get from Rx bus
+            data.delegate = await this.rxBus.get('DELEGATE_GET', req);
```
Comment (Verbalist) [inline api/controller/dashboard.ts]: This all data get from repository ACCOUNT or TRANSACTION not from bus

File: shared/model/dashboard.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    account: Account;
+    transactions: { data: Array<Transaction<Object>>; count: number };
+    unconfirmedTransaction?: { count: any; data: Array<Transaction<Object>> };
+    countStakeholders: number;
+    accountsCount: number;
+    delegate: Delegate;
+    forgingStatus: { enabled?: boolean; delegates: any };
+    totalDDKStaked: number;
+
+    constructor(rawData?: any) {
```
Comment (alex-arugon) [inline shared/model/dashboard.ts]: default value = {}

File: shared/model/dashboard.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        };
+        if (rawData.unconfirmedTransaction) {
+            this.unconfirmedTransaction = {
+                count: rawData.unconfirmedTransaction.count || 0,
+                data: rawData.unconfirmedTransaction.data || []
+            };
+        }
+        this.countStakeholders = Number(rawData.countStakeholders) || 0;
+        this.accountsCount = Number(rawData.accountsCount);
+        this.delegate = rawData.delegate || [];
```
Comment (alex-arugon) [inline shared/model/dashboard.ts]: default value 0

File: shared/model/dashboard.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        this.forgingStatus = rawData.forgingStatus; // todo implement from David
+        this.totalDDKStaked = Number(rawData.totalDDKStaked);
```
Comment (alex-arugon) [inline shared/model/dashboard.ts]: default value 0

**Pull request merged**